subsys/mpsl/fem/common/include/mpsl_fem_twi_drv.h

```diff
@@ -9,25 +9,43 @@
 
 #include <zephyr/devicetree.h>
 #include <mpsl_fem_nrf22xx_twi_config_common.h>
+#include <nrfx_twim.h>
 
-/**
- * @brief Initializer of structure @c mpsl_fem_twi_if_t allowing to use TWI interface.
- *
- * @param node_id Devicetree node identifier for the TWI part for the FEM device.
- */
-#define MPSL_FEM_TWI_DRV_IF_INITIALIZER(node_id) \
-	(mpsl_fem_twi_if_t){ \
-		.enabled = true, \
-		.slave_address = ((uint8_t)DT_REG_ADDR(node_id)), \
-		.p_instance = (void *)DEVICE_DT_GET(DT_BUS(node_id)), \
-		.p_xfer_read = mpsl_fem_twi_drv_impl_xfer_read, \
-		.p_xfer_write = mpsl_fem_twi_drv_impl_xfer_write, \
+/** @brief Structure representing an I2C bus driver for a Front-End Module. */
+typedef struct {
+	const struct device *dev;
+	nrfx_twim_evt_handler_t nrfx_twim_callback_saved;
+	void *nrfx_twim_callback_ctx_saved;
+	mpsl_fem_twi_async_xfer_write_cb_t fem_twi_async_xfwr_write_cb;
+	void *fem_twi_async_xfwr_write_cb_ctx;
+} mpsl_fem_twi_drv_t;
+
+#define MPSL_FEM_TWI_DRV_INITIALIZER(node_id)			\
+	(mpsl_fem_twi_drv_t){					\
+		.dev = DEVICE_DT_GET(DT_BUS(node_id)),		\
 	}
 
-int32_t mpsl_fem_twi_drv_impl_xfer_read(void *p_instance, uint8_t slave_address,
-			uint8_t internal_address, uint8_t *p_data, uint8_t data_length);
+/** @brief Prepares the FEM TWI interface structure.
```

Comment:
```suggestion
/** @brief Prepare the FEM TWI interface structure.
```
See https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/dev_model_and_contributions/documentation/styleguide.html#functions

Reply:
fixed